### PR TITLE
Fix `from_timestamp_millis` implementation and add more tests

### PR DESCRIPTION
**Pull request by esheppa**
Bug reported here: https://github.com/chronotope/chrono/issues/884

### CODE REVIEW

File: src/naive/datetime/tests.rs [hunk 1]
```diff
@@ -1,9 +1,37 @@
 use super::NaiveDateTime;
 use crate::oldtime::Duration;
 use crate::NaiveDate;
-use crate::{Datelike, FixedOffset, Utc};
+use crate::{Datelike, FixedOffset, Timelike, Utc};
 use std::i64;
 
+#[test]
+fn test_datetime_from_timestamp_millis() {
+    let timestamp_millis: i64 = 1662921288000; //Sunday, September 11, 2022 6:34:48 PM
```
Comment (djc) [inline src/naive/datetime/tests.rs:9]: So neither of the test cases you added have non-0 subsecond millis. Maybe add some cases for that so we have actual coverage?

File: src/naive/datetime/tests.rs [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+    let naive_datetime = NaiveDateTime::from_timestamp_millis(timestamp_millis);
+    assert!(naive_datetime.is_some());
+    assert_eq!(timestamp_millis, naive_datetime.unwrap().timestamp_millis());
+    assert_eq!(naive_datetime.unwrap().year(), 2022);
+    assert_eq!(naive_datetime.unwrap().month(), 9);
+    assert_eq!(naive_datetime.unwrap().day(), 11);
+    assert_eq!(naive_datetime.unwrap().hour(), 18);
+    assert_eq!(naive_datetime.unwrap().minute(), 34);
+    assert_eq!(naive_datetime.unwrap().second(), 48);
+    assert_eq!(naive_datetime.unwrap().timestamp_subsec_nanos(), 0);
```
Comment (djc) [inline src/naive/datetime/tests.rs:19]: Maybe cover most of the fields with a single `assert_eq!()` against a string-encoded timestamp? Would be a little more compact and readable.

**Review by djc — COMMENTED** [review]
Thanks for fixing this!

File: src/naive/datetime/tests.rs (hunk 1 above)
Comment (esheppa) [reply src/naive/datetime/tests.rs:9]: good call, I've added in 0 as well as MIN, MAX and some other interesting cases (these do a good test of `from_timestamp_opt` as well

File: src/naive/datetime/tests.rs (hunk 2 above)
Comment (esheppa) [reply src/naive/datetime/tests.rs]: good call - I've switched to comparing against a formatted string

File: src/naive/datetime/tests.rs [hunk 3]
```diff
@@ -4,6 +4,91 @@ use crate::NaiveDate;
 use crate::{Datelike, FixedOffset, Utc};
 use std::i64;
 
+#[test]
+fn test_datetime_from_timestamp_millis() {
+    let timestamp_millis: i64 = 1662921288000; //Sunday, September 11, 2022 6:34:48 PM
```
Comment (djc) [inline src/naive/datetime/tests.rs:9]: Okay now there's kind of a lot of duplication, though. Can we extract the test data out into a slice and loop over that?

Also doesn't seem useful to assert `is_some()` if the next line is going to `unwrap()` anyway?

Comment (esheppa) [reply src/naive/datetime/tests.rs]: Good point - used a map with a loop, and removed the assert some

**Review by djc — APPROVED** [review]

**Comment by esheppa** [comment]
Just fixed the issue on MSRV, you may want to re-review

**Review by djc — APPROVED** [review]

**Pull request merged**